### ISSUE TITLE
SIM116 detected twice

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#3257]
```python
def foo(func_name):
    if func_name == "create":
        return "A"
    elif func_name == "modify":
        return "M"
    elif func_name == "remove":
        return "D"
    elif func_name == "move":
        return "MV"
```

`SIM116` is detected twice, once is enough with highlighting the whole if-block:
```console
$ ruff --show-source foo.py
foo.py:2:5: SIM116 Use a dictionary instead of consecutive `if` statements
  |
2 |       if func_name == "create":
  |  _____^
3 | |         return "A"
4 | |     elif func_name == "modify":
5 | |         return "M"
6 | |     elif func_name == "remove":
7 | |         return "D"
8 | |     elif func_name == "move":
9 | |         return "MV"
  | |___________________^ SIM116
  |

foo.py:4:5: SIM116 Use a dictionary instead of consecutive `if` statements
  |
4 |       elif func_name == "modify":
  |  _____^
5 | |         return "M"
6 | |     elif func_name == "remove":
7 | |         return "D"
8 | |     elif func_name == "move":
9 | |         return "MV"
  | |___________________^ SIM116
  |

Found 2 errors.
```
